utils/testgen.py

```diff
@@ -359,6 +362,10 @@ def pytest_generate_tests(metafunc):
             if not(prov_obj.type_tclass in provider_types or prov_obj.type_name in provider_types):
                 continue
 
+        if provider_keys is not None:
```

Comment:
I presume this was so that we could use a provider key in the test to target a specific provider? However this is not something we would want to do. The key could change or be absent entirely.